src/core/lib/channel/promise_based_filter.cc

```diff
@@ -1167,13 +1938,36 @@ Poll<ServerMetadataHandle> ServerCallData::PollTrailingMetadata() {
   GPR_UNREACHABLE_CODE(return Pending{});
 }
 
+void ServerCallData::RecvTrailingMetadataReadyCallback(
+    void* arg, grpc_error_handle error) {
+  static_cast<ServerCallData*>(arg)->RecvTrailingMetadataReady(
+      std::move(error));
+}
+
+void ServerCallData::RecvTrailingMetadataReady(grpc_error_handle error) {
+  if (grpc_trace_channel.enabled()) {
+    gpr_log(GPR_DEBUG, "%s: RecvTrailingMetadataReady error=%s md=%s",
+            LogTag().c_str(), error.ToString().c_str(),
+            recv_trailing_metadata_->DebugString().c_str());
+  }
+  Flusher flusher(this);
+  PollContext poll_ctx(this, &flusher);
+  Cancel(error, &flusher);
```

Comment:
If the status is OK, then this isn't actually a cancellation.  Maybe rename `Cancel()` to just indicate that it's triggering completion, not necessarily failure?

Reply:
done